src/main/java/io/stargate/sgv2/jsonapi/service/operation/tasks/DBTask.java

```diff
@@ -0,0 +1,188 @@
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.api.model.command.table.definition.ColumnsDescContainer;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.CommandQueryExecutor;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.DefaultDriverExceptionHandler;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.DriverExceptionHandler;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import io.stargate.sgv2.jsonapi.util.CqlPrintUtil;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A single query we want to run against the database.
+ *
+ * <p>This can be any type of DML or DDL statement.
+ *
+ * <p>Subclasses are responsible for building the query to run, and how to handle the results. This
+ * super class knows how to be a task that is making a call to the Cassandra java driver.
+ *
+ * @param <SchemaT> The type of the schema object the task is working with.
+ */
+public abstract class DBTask<SchemaT extends SchemaObject>
+    extends BaseTask<SchemaT, DBTask.AsyncResultSetSupplier, AsyncResultSet> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DBTask.class);
+
+  protected final DefaultDriverExceptionHandler.Factory<SchemaT> exceptionHandlerFactory;
+
+  /**
+   * Create a new {@link DBTask} with the provided position, schema object, retry policy and
+   * exception handler factory.
+   *
+   * @param position See {@link BaseTask#BaseTask(int, SchemaObject, TaskRetryPolicy)}
+   * @param schemaObject See {@link BaseTask#BaseTask(int, SchemaObject, TaskRetryPolicy)}
+   * @param retryPolicy See {@link BaseTask#BaseTask(int, SchemaObject, TaskRetryPolicy)}
+   * @param exceptionHandlerFactory Factory to create a new {@link DriverExceptionHandler} that is
+   *     called when an exception is thrown after the driver retry policy.
+   */
+  protected DBTask(
+      int position,
+      SchemaT schemaObject,
+      TaskRetryPolicy retryPolicy,
+      DefaultDriverExceptionHandler.Factory<SchemaT> exceptionHandlerFactory) {
+    super(position, schemaObject, retryPolicy);
+
+    this.exceptionHandlerFactory =
+        Objects.requireNonNull(exceptionHandlerFactory, "exceptionHandlerFactory cannot be null");
+  }
+
+  // =================================================================================================
+  // BaseTask overrides
+  // =================================================================================================
+
+  /**
+   * A {@link UniSupplier} that for fetching a {@link AsyncResultSet}.
+   *
+   * <p>Includes the {@link SimpleStatement} so error mapping can include the CQL if needed. Note
+   * that the statement can change from request to request, e.g. if the first request missed an
+   * ALLOW FILTERING the next may have it.
+   */
+  public static class AsyncResultSetSupplier implements BaseTask.UniSupplier<AsyncResultSet> {
```

Comment:
Would this make sense as main-level class instead of `public` inner one? Or is it only ever used with `DBTask`?

Reply:
It is only used with the  DBTask subclasses. 

In the next "Big PR (tm)" the EmbeddingTask does not use this result set supplier, nor does the RerankingTask